livre/core/data/Histogram.cpp

```diff
@@ -97,35 +107,49 @@ bool Histogram::isEmpty() const
 
 uint64_t Histogram::getSum() const
 {
-    const uint64_t* bins = getBins();
+    const uint64_t* bins = getBins().data();
     uint64_t sum = 0;
-    for( size_t i = 0; i < getBinsSize(); ++i )
+    for( size_t i = 0; i < getBins().size(); ++i )
         sum +=  bins[ i ];
 
     return sum;
 }
 
+Vector2f Histogram::getRange() const
+{
+    return Vector2f( getMin(), getMax( ));
+}
+
 double Histogram::getRatio( const size_t index ) const
```

Comment:
Not that is important, but every time you call this function you loop over all the histogram to compute the sum of all counts. Do you need to get the ratio of a single index or in general you need all of them (for example for the GUI), because for the latter case you could have a function that returns a std::vector<float> with the normalized values and also remove the loop that the caller needs.